playlist/forms.py

```diff
@@ -0,0 +1,34 @@
+from django import forms
+
+from playlist.models import Song, Playlist
+
+
+class PlaylistForm(forms.ModelForm):
+    """To Add/Edit Playlist
+    """
+
+    class Meta:
+        model = Playlist
+        fields = ['title', 'description']
+
+
+class SongForm(forms.ModelForm):
+    """To Add/Edit Song
+    """
+
+    class Meta:
+        model = Song
+        fields = ['title', 'link']
+
+    def __init__(self, *args, **kwargs):
+        self.user = kwargs.pop('user', None)
+        self.playlist = kwargs.pop('playlist', None)
+        return super(SongForm, self).__init__(*args, **kwargs)
+
+    def save(self):
+        instance = super(SongForm, self).save(commit=False)
+        if self.user is not None and self.playlist is not None:
```

Comment:
stop writing hack codes.
1. there is no way a user can add a song if he/she is not logged in. so no need to check it.
2. don't save the song is playlist is not provided. show an error message to the user